src/components/common/AnimatedNumber.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react'
+import { LayoutChangeEvent, Text, TextStyle, View, ViewStyle } from 'react-native'
+import Animated, { Easing, EasingFunction, useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated'
+
+const ANIMATION_DURATION_DEFAULT = 1000
+const NUMBERS = Array(10)
+  .fill(0)
+  .map((_, i) => i)
+
+export interface AnimatedNumberProps {
+  animateToNumber: string
+  animationDuration?: number
+  textStyle?: TextStyle
+  easing?: EasingFunction
+  style?: ViewStyle
+}
+
+export const AnimatedNumber = (props: AnimatedNumberProps): JSX.Element => {
+  const { animateToNumber, textStyle, animationDuration = ANIMATION_DURATION_DEFAULT, easing = Easing.inOut(Easing.quad), style } = props
+  const animateToNumbersArr: string[] = Array.from(animateToNumber, String)
+  const [numberHeight, setNumberHeight] = React.useState(0)
+
+  const setButtonLayout = (event: LayoutChangeEvent) => {
+    setNumberHeight(event.nativeEvent.layout.height)
+  }
+
+  return (
+    <>
+      {numberHeight !== 0 ? (
+        <View style={[style, { flexDirection: 'row' }]}>
+          {animateToNumbersArr.map((n, index) => {
+            return (
+              <AnimatedDigit
+                animationDuration={animationDuration}
+                key={index}
+                index={index}
+                easing={easing}
+                textStyle={textStyle}
+                digit={n}
+                numberHeight={numberHeight}
+              />
+            )
+          })}
+        </View>
+      ) : (
+        <Text style={[textStyle, { position: 'absolute', top: -999999 }]} onLayout={setButtonLayout}>
+          0
+        </Text>
+      )}
+    </>
+  )
+}
+
+function isInteger(str: string) {
+  // Use parseInt with radix 10 to convert the string to an integer.
+  // If it's a valid integer, the result will not be NaN.
+  return !isNaN(parseInt(str, 10))
```

Comment:
`0.` is not NaN, so it will pass. I see we're only using this with one character, so perhaps we should document that for this function and avoid the `0xyz` cases. I suggest a rename `isDigit` and then a comment explaining the str should be length 1. Perhaps we could add a `.length` check?